### PR TITLE
chore: quote expiration cannot be set past the receiver's expiration

### LINKED ISSUE #593: Do not set quote expiration past receiver expiration
Use receiver's `expiresAt` if it is sooner than `quoteLifespan`
https://github.com/interledger/rafiki/blob/5e08158f445fca5fd8fd640bf0a66a71fd475993/packages/backend/src/open_payments/quote/service.ts#L339

### PR DESCRIPTION
**Pull request by SarahCoilAccount**
## Changes proposed in this pull request

- In cases where the receiver is an incoming payment it wouldn't make sense for the quotation's expiration to be set past the expiration date of the incoming payment.

## Context

- Fixes #593

## Checklist

- [x] Related issues linked using `fixes #number`
- [x] Tests added/updated
- [ ] Documentation added
- [ ] Make sure that all checks pass

### CODE REVIEW

**Comment by SarahCoilAccount** [comment]
When doing this work I had issues using the Rafiki GraphQL interface to create a quote. I tracked down the problem in the getIncomingPayment function in client/services.ts. I logged the error being caught there and it seems to still be jest related.

```
ReferenceError: jest is not defined
local-backend-1  |     at createContext (/workspace/node_modules/.pnpm/@shopify+jest-koa-mocks@5.0.1/node_modules/@shopify/jest-koa-mocks/build/cjs/create-mock-context/create-mock-context.js:29:22)
local-backend-1  |     at createReadContext (/workspace/packages/backend/dist/open_payments/client/service.js:121:78)
local-backend-1  |     at getIncomingPayment (/workspace/packages/backend/dist/open_payments/client/service.js:168:25)
local-backend-1  |     at runMicrotasks (<anonymous>)
local-backend-1  |     at processTicksAndRejections (node:internal/process/task_queues:96:5)
local-backend-1  |     at async getReceiver (/workspace/packages/backend/dist/open_payments/client/service.js:209:33)
```

File: packages/backend/src/open_payments/client/service.ts [hunk 1]
```diff
@@ -45,14 +45,16 @@ export class Receiver extends ConnectionBase {
...
+      expiryDate
     )
   }
 
   private constructor(
     connection: ConnectionJSON,
     private readonly incomingAmountValue?: bigint,
-    private readonly receivedAmountValue?: bigint
+    private readonly receivedAmountValue?: bigint,
+    private readonly expiresAtValue?: Date
```
Comment (wilsonianb) [inline packages/backend/src/open_payments/client/service.ts:57]:
```suggestion
    public readonly expiresAt?: Date
```
I don't think this needs the getter below

Comment (wilsonianb) [reply packages/backend/src/open_payments/client/service.ts:57]: Can you update client service / receiver tests to incorporate `expiresAt`?
https://github.com/interledger/rafiki/blob/2c40d222006e5beef2fe850d0208c27c70bfcc69/packages/backend/src/open_payments/client/service.test.ts

**Review by wilsonianb — COMMENTED** [review]
I wasn't expecting this to affect the client service, but it makes sense.
There'll likely be conflicts between this and:
- https://github.com/interledger/rafiki/pull/706

File: packages/backend/src/open_payments/quote/service.ts [hunk 2]
```diff
@@ -271,6 +271,11 @@ export async function finalizeQuote(
     'Content-Type': 'application/json'
   }
 
+  const receiver = await deps.clientService.receiver.get(quote.receiver)
```
Comment (wilsonianb) [inline packages/backend/src/open_payments/quote/service.ts:274]: We shouldn't need to make another request to get the receiver here.
We could instead pass `receiver` as an argument to `finalizeQuote`.

**Review by wilsonianb — APPROVED** [review]

**Pull request merged**